src/main/java/org/igniterealtime/openfire/botz/BotzConnection.java

```diff
@@ -175,7 +177,19 @@ public void deliverRawText(String text) {
 		packetReceiver.processIncomingRaw(text);
 	}
 
-	/*
+    @Override
+    public Optional<String> getTLSProtocolName() {
+        return Optional.empty();
```

Comment:
Why not do the same thing as with CipherSuiteName: check if the session is non-null, and then delegate to that.